architecture.md

```diff
@@ -0,0 +1,91 @@
+Other such libraries include [Ignition Plugin](https://ignitionrobotics.org/libs/plugin), [Ignition Math](https://ignitionrobotics.org/libs/math), [SDFormat](https://ignitionrobotics.org/libs/sdformat), and more.
+
+There are certain plugins in both the frontend and backend processes that are loaded by default in every version of Ignition.
+Many other plugins are optional.
+One common optional plugin is the sensors plugin.
+Both optional and default plugins can be removed or added at any time; Ignition Gazebo will continue to run with limited functionality.
+These demos on [Server Configuration](https://ignitionrobotics.org/api/gazebo/4.3/server_config.html) and [GUI Configuration](https://ignitionrobotics.org/api/gazebo/4.3/gui_config.html) showcase that functionality.
+
+The simulation process is depicted in the diagram below, and further explained in the Backend and Frontend process sections that follow.
+
+![Ignition Gazebo architecture diagram](images/IgnitonGazeboArchicteture.png)
+
+## Backend server process
+
+Ignition Gazebo is responsible for loading plugins in the backend, referred to as systems.
+The server runs an entity-component system architecture (see [Ignition Gazebo terminology](https://ignitionrobotics.org/api/gazebo/4.2/terminology.html)).
+The backend will usually have multiple systems responsible for everything in the simulation – computing physics, recording logs, receiving user commands, etc.
+
+Systems act on entities and components of those entities.
+An entity is anything in the simulation scene (a model, link, light, actor, etc.), and components are their characteristics (pose, name, geometry, etc.).
+Server plugins have access to all entities and what components they have, and make modifications to those components.
+For example, a user could write a system that applies force to an entity by setting a component on that entity, and the physics system would pick that up and react by applying the force to make the entity actually move.
+The modification of entities and components is how server plugins communicate with each other.
+Other Ignition Libraries, however, don’t know about the entities and components.
+
+There is a loop running in the backend that runs the systems, where some systems are proposing changes to entities and their components, and other systems are handling and applying those changes.
+This is called the “simulation loop”.
+The Entity Component Manager (ECM) in the backend provides the functionality for the actual querying and updating of the entities and components.
+
+Physics, user commands, and scene broadcaster are all systems launched by default in the backend.
+As mentioned earlier, however, even default systems can be added to or removed from this loop.
```

Comment:
```suggestion
Physics, user commands, and scene broadcaster are all systems launched by default in the backend.
As mentioned earlier, however, even default systems can be added to or removed from the simulation loop (the [Server Configuration](https://ignitionrobotics.org/api/gazebo/4.3/server_config.html) tutorial describes how to customize default systems).
```

I think it would be good to provide users with more information about adding/removing default systems in case they're curious.

I also replaced "this loop" with "simulation loop" for clarity (users may not recall what "this loop" is since this is a new paragraph).